sumo/io/castep.py

```diff
@@ -12,9 +12,13 @@
 from pymatgen.core.structure import Structure
 from pymatgen.electronic_structure.core import Spin
 from pymatgen.electronic_structure.bandstructure import BandStructureSymmLine
-from pymatgen.electronic_structure.dos import Dos
+from pymatgen.electronic_structure.dos import Dos, CompleteDos
 from pymatgen.phonon.bandstructure import PhononBandStructureSymmLine
 
+from sumo.electronic_structure.dos import get_pdos
+
+from castepxbin.pdos import compute_pdos
```

Comment:
Realistically this import is going to fail sometimes. I guess our options are:

1. make this a fully optional dependency, and fall back to TDOS with a warning message if it is not installed
2. make this a required dependency _for CASTEP users_ and provide a more helpful error message if this import fails
3. make this part of the standard sumo installation (as you have done so far in the PR). In which case the default error is ok

My current preference would be (1) but I'd be ok with (2) if the castepxbin package is on PyPI. (Not keen on github imports in setup.py files.) I remain open to being persuaded that (3) is the way forward especially if @utf agrees.

Reply:
(3) can be taken as now the external code is released on pypi

Reply:
As castepxbin is a small package without onerous dependencies I am happy to accept it as a mandatory dependency of Sumo at this stage. However, I note that this approach will not be suitable for all code interfaces; if we start using ASE that should probably be optional.